src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,115 @@
+    private static final Logger logger = LoggerFactory.getLogger(ReflectionTest.class);
+
+    @Test
+    @DisplayName("Car 객체 정보 가져오기")
+    void showClass() {
+        Class<Car> carClass = Car.class;
+        logger.debug(carClass.getName());
+
+        // 필드
+        Field[] fields = carClass.getDeclaredFields();
+        System.out.println("필드:");
+        for (Field field : fields) {
+            System.out.println(field);
+        }
+
+        // 생성자
+        Constructor<?>[] constructors = carClass.getDeclaredConstructors();
+        System.out.println("생성자:");
+        for (Constructor<?> constructor : constructors) {
+            System.out.println(constructor);
+        }
+
+        // 메서드
+        Method[] methods = carClass.getDeclaredMethods();
+        System.out.println("메서드");
+        for (Method method : methods) {
+            System.out.println(method);
+        }
+    }
+
+    @Test
+    @DisplayName("test 로 시작하는 메소드 실행")
+    void testMethodRun() throws InstantiationException, IllegalAccessException, InvocationTargetException {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] constructors = carClass.getDeclaredConstructors();
+        Method[] methods = carClass.getDeclaredMethods();
+        for (Method method : methods) {
+            String s = method.getName();
+            if (s.startsWith("test")) {
+                Object invoke = method.invoke(constructors[0].newInstance());
+                System.out.println(invoke);
+            }
+        }
+    }
+
+    @Test
+    @DisplayName("@PrintView 애노테이션 메소드 실행")
+    void testAnnotationMethodRun() throws InvocationTargetException, InstantiationException, IllegalAccessException {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] constructors = carClass.getDeclaredConstructors();
+        Method[] methods = carClass.getDeclaredMethods();
+        for (Method method : methods) {
+            String s = method.getName();
+            if (method.isAnnotationPresent(PrintView.class)) {
+                method.invoke(constructors[0].newInstance());
+            }
+        }
+    }
+
+
+    @Test
+    @DisplayName("private field에 값 할당")
+    public void privateFieldAccess() throws InvocationTargetException, InstantiationException, IllegalAccessException {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] constructors = carClass.getDeclaredConstructors();
+        Car car = (Car) constructors[0].newInstance();
+
+        // 필드
+        Field[] fields = carClass.getDeclaredFields();
+        for (Field field : fields) {
+            field.setAccessible(true);
+            if (field.getName().equals("name")) {
+                field.set(car, "소나타");
+            } else if (field.getName().equals("price")) {
+                field.set(car, 123);
+            }
+        }
+
+        assertThat(car.testGetName()).isEqualTo("test : 소나타");
+        assertThat(car.testGetPrice()).isEqualTo("test : 123");
```

Comment:
```suggestion
        assertAll(
                () -> assertThat(car.testGetName()).isEqualTo("test : 소나타"),
                () -> assertThat(car.testGetPrice()).isEqualTo("test : 123")
        );
```
자바에서 `assertThat`을 단일로 사용하면 assert문이 실패할 경우 아래 메소드는 실행하지 않기 때문에 112가 실패하면 113은 실패한지 아닌지 확인할 수 없어 여러 assert문을 검증하는 경우 `assertAll`를 사용하는 것이 좋습니다 :)